### PR TITLE
techdocs: fix buildLogTransport-related memory leak

### PR DESCRIPTION
**Pull request by zcmander**
This PR is a naive fix for the memory leak issue described in: #20698

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by zcmander** [comment]
not stale, just waiting for review...

**Review by freben — APPROVED** [review]
Alright, let's try it. Ping @backstage/techdocs-maintainers again FYI though

**Pull request merged**
